static/app/utils/useNavigate.tsx

```diff
@@ -1,28 +1,43 @@
-import {useContext} from 'react';
+import {useCallback, useEffect, useRef} from 'react';
 
-import {RouteContext} from 'sentry/views/routeContext';
+import {useRouteContext} from 'sentry/utils/useRouteContext';
 
 type NavigateOptions = {
   replace?: boolean;
   state?: any;
 };
 
 export function useNavigate() {
-  const route = useContext(RouteContext);
-  if (route === null) {
-    throw new Error('useNavigate called outside of routes provider');
-  }
-  const navigator = route?.router;
+  const route = useRouteContext();
 
-  const navigate = (to: string | number, options: NavigateOptions = {}) => {
-    if (typeof to === 'number') {
-      navigator.go(to);
-      return;
-    }
-    (!!options.replace ? navigator.replace : navigator.push)({
-      pathname: to,
-      state: options.state,
-    });
-  };
+  const navigator = route.router;
+  const activeRef = useRef(false);
```

Comment:
Nit, but dont think active is exactly correct here, we are checking if the component has mounted, so maybe something like
```suggestion
  const hasMountedRef = useRef(false);
```